src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -614,12 +936,11 @@ QMCCostFunctionBatched::EffectiveWeight QMCCostFunctionBatched::correlatedSampli
 
         TrialWaveFunction::mw_evaluateParameterDerivatives(wf_list, p_list, optVars, dlogpsi_array, dhpsioverpsi_array);
 
-
-        // Energy
         auto energy_list =
-            QMCHamiltonian::mw_evaluateValueAndDerivatives(h0_list, wf_list, p_list, optVars, dlogpsi_array,
+            QMCHamiltonian::mw_evaluateValueAndDerivatives(h_list, wf_list, p_list, optVars, dlogpsi_array,
```

Comment:
This change from h0 to h is a change in behavior of existing code, and changes how the batched code works relative to the legacy code.   If this was an intentional change, it needs to be understood more deeply.
"h0_list" in the batched code corresponds to "H_KE_Node" in the legacy code.  It includes the KE energy part of the Hamiltonian and, optionally, the nonlocal part of the psuedopotential ("nonlocalpp" parameter in the input file)

Reply:
Presumably this change is contributing to the broken optimizer tests revealed in CI.